src/main/java/edu/kis/vh/nursery/list/IntLinkedList.java

```diff
@@ -1,22 +1,58 @@
 package edu.kis.vh.nursery.list;
 
-public class IntLinkedList {
+class IntLinkedList {
 
-	Node last;
-	int index;
+	static final int EMPTY = -1;
+	
+	class Node {
+
+		private final int value;
+		private Node prev;
+		private Node next;
+
+		public Node(int value) {
+			this.value = value;
+		}
+
+		public int getValue() {
+			return value;
+		}
+
+		public Node getPrev() {
+			return prev;
+		}
+
+		public void setPrev(Node prev) {
+			this.prev = prev;
+		}
+
+		public Node getNext() {
+			return next;
+		}
+
+		public void setNext(Node next) {
+			this.next = next;
+		}
+		
+		/*
+		 * W innych klasach ustawianie wartosci zostalo zamienione na uzycie settera a pobieranie na uzycie gettera
+		 */
+	}
+
+	private Node last;

```

Comment:
'last' shouldn't be encapsulated (typical private field). This is an exaggeration. Using private fields internallt usually do not require encapsulation.